### PR TITLE
Fix generation of UART module for AVRs without specified instances

### LINKED ISSUE #206: UART on ATmega8
I'm trying to write a program for an ATmega8 that needs UART.

Based on the AVR examples I put together this `project.xml`:
```xml
<library>
  <options>
    <option name="modm:target">atmega8</option>
    <option name="modm:platform:clock:f_cpu">8000000</option>
    <option name="modm:io:with_printf">True</option>
    <option name="modm:build:avrdude.programmer">avrisp2</option>
  </options>
  <modules>
    <module>modm:architecture:interrupt</module>
    <module>modm:driver:mcp2515</module>
    <module>modm:platform:clock</module>
    <module>modm:platform:core</module>
    <module>modm:platform:gpio</module>
    <module>modm:platform:spi</module>
    <module>modm:platform:uart:0</module>
    <module>modm:processing:timer</module>
    <module>modm:build:scons</module>
  </modules>
</library>
```
When I run `lbuild -r ..\modm\repo.lb build` I get this output:
```
Recomputing device cache...

ERROR: Cannot resolve 'modm:platform:uart:0'
```

I tried replacing `modm:platform:uart:0` with `modm:platform:uart`, `modm:platform:uart:1`, but nothing seems to work. Does this mean that the UART is not supported on the mega8 yet?

### PR DESCRIPTION
**Pull request by Sh4rK**
Fixes #206.

### CODE REVIEW

File: src/modm/platform/uart/at90_tiny_mega/module.lb [hunk 1]
```diff
@@ -84,20 +84,28 @@ def prepare(module, options):
...
         ":platform:gpio")
 
-    driver = device.get_driver("usart")
+    driver = device.get_driver("usart:avr")
     if "instance" in driver:
         for instance in listify(driver["instance"]):
             module.add_submodule(Instance(int(instance)))
     else:
-        return False
+        load_options(module)
```
Comment (Sh4rK) [inline src/modm/platform/uart/at90_tiny_mega/module.lb:92]: Disclaimer: I don't know what this does, I just copied it from the SPI implementation you pointed at.

**Comment by Sh4rK** [comment]
Should I add a test exercising this? If so, where?

File: src/modm/platform/uart/at90_tiny_mega/module.lb (hunk 1 above)
Comment (salkinium) [reply src/modm/platform/uart/at90_tiny_mega/module.lb:92]: That calls the [function defined in L22](https://github.com/modm-io/modm/blob/develop/src/modm/platform/uart/at90_tiny_mega/module.lb#L22) that just adds the Buffer options. It's just a separate function so you can use it for both the `modm:platform:uart` as well as its submodules.

**Comment by salkinium** [comment]
> Should I add a test exercising this? If so, where?

You can add a simple example using the UART with the `atmega8` as target in the `example/avr` folder, which then becomes part of the CI and remains maintained by me. Almost all of them are without board files, just "raw" AVR targets.

Basically copy this one and use `Uart` instead of Uart0` for `atmega8`.
https://github.com/modm-io/modm/blob/develop/examples/avr/uart/basic/main.cpp

**Comment by Sh4rK** [comment]
Do I need to add the example to any scripts or config files for it to get compiled on CI?

**Comment by Sh4rK** [comment]
```
#!/bin/bash -eo pipefail
python3 tools/scripts/synchronize_docs.py -d
git diff
diff --git a/examples/README.md b/examples/README.md
index 992690c87..73183a320 100644
--- a/examples/README.md
+++ b/examples/README.md
@@ -94,7 +94,7 @@ make gdb
 
 ## Interesting Examples
 
-We have a lot of examples, <!--examplecount-->170<!--/examplecount--> to be
+We have a lot of examples, <!--examplecount-->171<!--/examplecount--> to be
 exact, but here are some of our favorite examples for our supported development
 boards:
 

Please synchronize the modm documentation:

    $ python3 tools/scripts/synchronize_docs.py

 and then commit the results!
```

This is pretty cool :)

EDIT: And also, surprise, does not work on windows. I updated the example count manually.

**Comment by salkinium** [comment]
> Do I need to add the example to any scripts or config files for it to get compiled on CI?

No, the `tools/scripts/example_compile.py` scripts searches for all `project.xml` files in `avr/` and compiles them. You only need to add new top-level folders in `examples/`. This is so we can parallelize example compilation and use the 4x containers in Circle CI.

> This is pretty cool :)

Yeah, and now I can't forget to update the docs anymore. I mean, contributers can't forget… \*cough\* ;-P

> EDIT: And also, surprise, does not work on windows.

I've put a Windows CI on my todo list. We only need to compile a few typical examples to fix these very low-hanging ~~fruits~~ bugs…

**Review by salkinium — DISMISSED** [review]
Thanks!

**Pull request merged**